pyomo/gdp/plugins/bigm.py

```diff
@@ -231,9 +237,32 @@ def _apply_to_impl(self, instance, **kwds):
         targets = config.targets
         if targets is None:
             targets = (instance, )
-        # We need to check that all the targets are in fact on instance. As we
-        # do this, we will use the set below to cache components we know to be
-        # in the tree rooted at instance.
+        else:
+            # we need to preprocess targets to make sure that if there are any
+            # disjunctions in targets that their disjuncts appear before them in
+            # the list.
+            preprocessed_targets = []
+            for t in targets:
+                if t.ctype is Disjunction:
+                    if t.is_indexed():
+                        for disjunction in t.values():
+                            for disj in disjunction.disjuncts:
+                                print("adding Disjunct %s" % disj.name)
+                                preprocessed_targets.append(disj)
+                    else:
+                        for disj in t.disjuncts:
+                            print("adding Disjunct %s" % disj.name)
+                            preprocessed_targets.append(disj)
+                # now we are safe to put the disjunction, and if the target was
+                # anything else, then we don't need to worry because disjuncts
+                # are declared before disjunctions they appear in
+                print("adding target %s of type %s" % (t.name, t.ctype))
```

Comment:
Is this debugging? If this is expected output, please convert to using the logger.